website/src/components/Chat/ChatConversation.tsx

```diff
@@ -165,23 +170,61 @@ export const ChatConversation = memo(function ChatConversation({ chatId: chatIdP
     [sendVote]
   );
 
-  const entries = useMemo(
-    () =>
-      currentThread.map((message) => (
-        <ChatMessageEntry
-          key={message.id}
-          message={message}
-          onVote={handleOnVote}
-          onRetry={handleOnRetry}
-          isSending={isSending}
-        ></ChatMessageEntry>
-      )),
-    [handleOnVote, currentThread, handleOnRetry, isSending]
+  const handleEditPrompt = useCallback(
+    async ({ chatId, parentId, content }: EditPromptParams) => {
+      if (!content || isSending) {
+        return;
+      }
+
+      setIsSending.on();
+      setReytryingParentId(parentId);
+      const prompter_arg: InferencePostPrompterMessageParams = {
+        chat_id: chatId,
+        content,
+        parent_id: parentId,
+      };
+
+      let prompter_message: InferenceMessage | null = null;
+      const dummyMessage: InferenceMessage = {
+        id: "__dummy__",
+        ...prompter_arg,
+        created_at: new Date().toISOString(),
+        role: "prompter",
+        state: "complete",
+        score: 0,
+        reports: [],
+      };
+
+      try {
+        // push the dummy message first to avoid layout shift
+        setMessages((messages) => [...messages, dummyMessage]);
+        prompter_message = await post(API_ROUTES.CREATE_PROMPTER_MESSAGE, { arg: prompter_arg });
+        // filter out the dummy message and push the real one
+        setMessages((messages) => [...messages.filter((m) => m.id !== "__dummy__"), prompter_message!]);
+      } catch {
+        // revert on any error
+        setMessages((messages) => [...messages.filter((m) => m.id !== "__dummy__")]);
```

Comment:
if an error happens, we arguably should not continue executing this function at all.

Also, I don't think handling the error gracefully is necessary, there are so many edge cases where stuff could go wrong in the entire app, and we don't handle any of them gracefully.

Reply:
> if an error happens, we arguably should not continue executing this function at all.

Yes we only continue the function if the prompter message is created successfully. The code below already handle this

```js
if (prompter_message) {
    await initiate_assistant_message({ parentId: prompter_message.id, chatId: chatId });
}
```